task_3/theater-gorm/pkg/data/theater.go

```diff
@@ -0,0 +1,646 @@
+		Joins("JOIN price on performance.id = price.performance_id and sectors.id = price.sector_id").
+		Rows()
+	if err != nil {
+		return nil, fmt.Errorf("can't read users from database, error:%w", err)
+	}
+	for rows.Next() {
+		temp := SelectTicket{}
+		err := rows.Scan(&temp.Id, &temp.PerformanceName, &temp.GenreName, &temp.PerformanceDuration, &temp.DateTime, &temp.HallName, &temp.HallCapacity, &temp.LocationAddress, &temp.LocationPhoneNumber, &temp.SectorName, &temp.Place, &temp.Price, &temp.DateOfIssue, &temp.Paid, &temp.Reservation, &temp.Destroyed)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan tickets from database, error:%w", err)
+		}
+		tickets = append(tickets, temp)
+	}
+	return tickets, nil
+}
+
+func (u TheaterData) ReadAllPosters() ([]SelectPoster, error) {
+	var posters []SelectPoster
+	rows, err := u.db.Table("poster").Select("poster.id, performance.name, genre.name, " +
+		"performance.duration, schedule.date, halls.name, halls.capacity, locations.address, locations.phone_number, poster.comment ").
+		Joins("JOIN schedule on schedule.id = poster.schedule_id").
+		Joins("JOIN performance on schedule.performance_id = performance.id").
+		Joins("JOIN genre on performance.genre_id = genre.id").
+		Joins("JOIN halls on schedule.hall_id = halls.id").
+		Joins("JOIN locations on halls.location_id = locations.id").
+		Rows()
+	if err != nil {
+		return nil, fmt.Errorf("can't get posters from database, error:%w", err)
+	}
+	for rows.Next() {
+		var temp SelectPoster
+		err = rows.Scan(&temp.Id, &temp.PerformanceName, &temp.GenreName, &temp.PerformanceDuration,
+			&temp.DateTime, &temp.HallName, &temp.HallCapacity, &temp.LocationAddress, &temp.LocationPhoneNumber,
+			&temp.Comment)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan posters from database, error:%w", err)
+		}
+		posters = append(posters, temp)
+	}
+	return posters, nil
+}
+
+func (u TheaterData) ReadAllUsers(account Account) ([]SelectUser, error) {
+	var users []SelectUser
+	rows, err := u.db.Table("users").Select("users.id, users.first_name, "+
+		"users.last_name, roles.name, locations.address, locations.phone_number, users.phone_number").
+		Joins("JOIN roles on users.role_id = roles.id").
+		Joins("JOIN locations on locations.id = users.account_id").
+		Where("users.account_id = ?", account.Id).
+		Rows()
+	if err != nil {
+		return nil, fmt.Errorf("can't get users from database, error:%w", err)
+	}
+	for rows.Next() {
+		var temp SelectUser
+		err = rows.Scan(&temp.Id, &temp.FirstName, &temp.LastName, &temp.Role,
+			&temp.LocationAddress, &temp.LocationPhoneNumber, &temp.PhoneNumber)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan users from database, error:%w", err)
+		}
+		users = append(users, temp)
+	}
+	return users, nil
+}
+
+func (u TheaterData) AddAccount(account Account) (int, error) {
+	result := u.db.Create(&account)
+	if result.Error != nil {
+		return -1, fmt.Errorf("can't inser account to database, error: %w", result.Error)
```

Comment:
also here, replace inser to insert in all places

Reply:
Done